### PR TITLE
Upgrade Scala to 2.12.8 and all the dependencies.

### PR DESCRIPTION
**Pull request by snadorp**
Upgrade Scala to 2.12.8 and also upgrade cats-core, cats-effect, circe and specs2

### CODE REVIEW

**Comment by snadorp** [comment]
Closes #39 and #40

**Comment by snowplowcla** [comment]
Confirmed! @snadorp  has signed the Individual Contributor License Agreement. Thanks so much

File: project/Dependencies.scala [hunk 1]
```diff
@@ -13,24 +13,23 @@
...
-
 import sbt._
 
 object Dependencies {
   object V {
-    val catsCore = "1.1.0"
-    val catsEffect = "0.10.1"
-    val circe = "0.10.0"
-    val specs2 = "4.2.0"
+    val catsCore   = "1.4.0"
```
Comment (BenFradet) [inline project/Dependencies.scala]: the latest version is 1.5.0 I think

File: project/Dependencies.scala [hunk 2]
```diff
@@ -13,24 +13,23 @@
...
+    val catsEffect = "1.0.0"
```
Comment (BenFradet) [inline project/Dependencies.scala]: same I think it's 1.1.0

**Comment by snadorp** [comment]
@BenFradet done.

**Pull request merged**